### PR TITLE
fix: do not ever skip updates which have remove flag

### PR DESCRIPTION
**Pull request by Unix4ever**
For remove we don't care about what's set in the current peers. Otherwise it leads to remove event being skipped if the same values for the address or keep alive interval are passed.

### CODE REVIEW

File: pkg/wireguard/wireguard.go [hunk 1]
```diff
@@ -552,6 +552,11 @@ func PrepareDeviceConfig(peerEvents []PeerEvent, oldCfg *wgtypes.Device, userHan
 
 func checkDuplicateUpdates(seq iter.Seq[PeerEvent], oldCfg *wgtypes.Device, logger *zap.Logger) iter.Seq[PeerEvent] {
 	return iter.Filter(seq, func(peerEvent PeerEvent) bool {
+		// never skip remove
```
Comment (smira) [inline pkg/wireguard/wireguard.go:555]: what if that's add and remove later on?

my point we should dedup in some more intelligent way?

Comment (smira) [reply pkg/wireguard/wireguard.go:555]: or... in other terms what if the sequence of events is like:

`add` -> `remove` -> `add` ?

Comment (Unix4ever) [reply pkg/wireguard/wireguard.go:555]: do you mean `update` -> `update` -> `remove` should turn into a single `remove`?
That's already ok and is covered by the test case.

Comment (smira) [reply pkg/wireguard/wireguard.go:555]: I mean exactly what I posted above, as I think 3rd `add` will get de-duplicated, while `remove` will be executed. 

If we remove, we should only execute it if the peer is there... right?

Comment (smira) [reply pkg/wireguard/wireguard.go:555]: Or is there additional step when we combine same events for the same pubkey?

Comment (Unix4ever) [reply pkg/wireguard/wireguard.go:555]: yep, there's. It sorts, then deduplicates

Comment (smira) [reply pkg/wireguard/wireguard.go:555]: so the proper check for 'remove' should be that it doesn't exist in the peers list?

Comment (smira) [reply pkg/wireguard/wireguard.go:555]: sorry, that it actually exists in the peer list

Comment (Unix4ever) [reply pkg/wireguard/wireguard.go:555]: I even checked that it works this way by writing the test case. Can commit it I guess.

Comment (Unix4ever) [reply pkg/wireguard/wireguard.go:555]: Yeah, good point. Will add that

Comment (Unix4ever) [reply pkg/wireguard/wireguard.go]: updated, should be good now.

**Review by smira — APPROVED** [review]

**Pull request merged**